test/webserver.mjs

```diff
@@ -120,83 +117,83 @@ class WebServer {
     }
 
     if (pathPart === "/favicon.ico") {
-      fs.realpath(
-        path.join(this.root, "test/resources/favicon.ico"),
-        checkFile
-      );
-      return;
+      pathPart = "test/resources/favicon.ico";
```

Comment:
*I might be missing something, however:*
In the old code we'd *only* call the `checkFile` function and then return early here, but it appears that we're now (potentially) running a bunch more code in this case?

Reply:
I was initially equally confused about the old code here, but it turns out the early return doesn't really achieve much. The `checkFile` call here in turn calls `statFile` which in turn serves the favicon file. However, note that the line below the early return at https://github.com/mozilla/pdf.js/blob/master/test/webserver.mjs#L133 does the exact same thing, just for a different path part, and the rest of the handler function below the early return doesn't actually do anything other than defining the inline functions and some variables. I also thought a lot more was happening here, but the inline functions made that hard to see.

Reply:
Oh, that's right; I was getting confused by the "callback hell" of the old code!

This just shows *why* these kind of patches are so worthwhile, since the use of async/await means the code can now be read from top to bottom without any trouble!